.github/workflows/safe-transaction.yml

```
@@ -0,0 +1,87 @@
+name: 'Safe Transaction Proposer'
+on:
+  workflow_call:
+    inputs:
+      rpc-url:
+        description: 'RPC URL for the blockchain network'
+        required: true
+        type: string
+      chain-id:
+        description: 'Chain ID of the blockchain network (default: 42161 for Arbitrum)'
+        required: false
+        default: '42161'
+        type: string
+      safe-address:
+        description: 'Address of the Safe contract'
+        required: true
+        type: string
+      transaction-to:
+        description: 'Target address for the transaction'
+        required: true
+        type: string
+      transaction-value:
+        description: 'Value to send in the transaction (in wei, default: 0)'
+        required: false
+        default: '0'
+        type: string
+      transaction-data:
+        description: 'Transaction data/calldata (default: 0x)'
+        required: false
+        default: '0x'
+        type: string
+    secrets:
+      proposer-private-key:
+        description: 'Private key of the proposer wallet'
+        required: true
+      safe-api-key:
+        description: 'Safe API key for transaction service'
+        required: true
+    outputs:
+      safe-tx-hash:
+        description: 'Hash of the Safe transaction'
+        value: ${{ jobs.propose-transaction.outputs.safe-tx-hash }}
+      transaction:
+        description: 'Created transaction details'
+        value: ${{ jobs.propose-transaction.outputs.transaction }}
+
+jobs:
+  propose-transaction:
+    runs-on: ubuntu-latest
+    outputs:
+      safe-tx-hash: ${{ steps.safe-transaction.outputs.safe-tx-hash }}
+      transaction: ${{ steps.safe-transaction.outputs.transaction }}
+    
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          repository: iExecBlockchainComputing/github-actions-workflows
+          # TODO: Only for testing, remove before merging
+          ref: feature/add-safe-transaction-reusable-workflow
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '22'
+
+      - name: Install dependencies
+        run: npm ci
+        working-directory: ./safe-transaction
+
+      - name: Build action
+        run: npm run build
+        working-directory: ./safe-transaction
+
+      - name: Propose Safe Transaction
+        run: node dist/index.js
```

```suggestion
        run: npm run propose
```
This makes the action independent of the project internal architecture.